moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -389,17 +389,15 @@ void PlanningScene::clearDiffs()
 
     if (it.second->parent_)
     {
-      it.second->cenv_ = it.second->alloc_->allocateEnv(it.second->parent_->cenv_, world_);
-      it.second->cenv_const_ = it.second->cenv_;
-
       it.second->cenv_unpadded_ = it.second->alloc_->allocateEnv(it.second->parent_->cenv_unpadded_, world_);
       it.second->cenv_unpadded_const_ = it.second->cenv_unpadded_;
+
+      it.second->cenv_ = it.second->alloc_->allocateEnv(it.second->parent_->cenv_, world_);
+      it.second->cenv_const_ = it.second->cenv_;
     }
-    else
+    else  // This is the parent CollisionDetector
     {
       it.second->copyPadding(*parent_->active_collision_);
-
-      it.second->cenv_ = it.second->alloc_->allocateEnv(it.second->parent_->cenv_, world_);
```

Comment:
problem here because `it.second->parent_` is null

Reply:
I'm not sure I understand. Why is this line deleted?

Reply:
If you look at line 390, `if (it.second->parent_)` --> the pointer is null, so this will segfault.

That also means there is no parent CollisionDetector, so copying the parent's collision environment doesn't make sense.

I figure the original author copy/pasted this from L395 but forgot to update it.